lib/src/rules/unnecessary_lambdas.dart

```diff
@@ -31,25 +31,37 @@ names.forEach(print);
 
 ''';
 
+bool _containsNullAwareInvocationInChain(AstNode node) =>
+    node != null &&
+    ((node is PropertyAccess &&
+            (node.operator?.type == TokenType.QUESTION_PERIOD ||
+                _containsNullAwareInvocationInChain(node.target))) ||
+        (node is MethodInvocation &&
+            (node.operator?.type == TokenType.QUESTION_PERIOD ||
+                _containsNullAwareInvocationInChain(node.target))) ||
+        (node is IndexExpression &&
+            _containsNullAwareInvocationInChain(node.target)));
+
 Iterable<Element> _extractElementsOfSimpleIdentifiers(AstNode node) =>
     DartTypeUtilities
         .traverseNodesInDFS(node)
         .where((e) => e is SimpleIdentifier)
         .map((e) => (e as SimpleIdentifier).bestElement);
 
-bool _hasRecursivelyQuestionPeriod(AstNode node) {
-  if (node == null) {
-    return false;
-  }
-  return (node is PropertyAccess &&
-          (node.operator?.type == TokenType.QUESTION_PERIOD ||
-              _hasRecursivelyQuestionPeriod(node.target))) ||
-      (node is IndexExpression &&
-              _hasRecursivelyQuestionPeriod(node.target)) ||
-      (node is MethodInvocation &&
-          (node.operator?.type == TokenType.QUESTION_PERIOD ||
-              _hasRecursivelyQuestionPeriod(node.target)));
-}
+bool _isInvocationExpression(AstNode node) => node is InvocationExpression;
+
+// invocation chain contains null aware invocation (?.)
```

Comment:
No, Alexis was helping me to chose better function names and I forgot to delete the comment, sorry for that.

Fixed